packages/material-ui-codemod/src/util/allowedPrivateImportsList.js

```diff
@@ -0,0 +1,30 @@
+const allowedPrivateImportsList = [
```

Comment:
I've created this list by searching for `@ignore - internal component`. Apart from these all components in `@material-ui/core/internals` are also allowed by the codemod.

Reply:
Why are these allowed? They are explicitly marked as internal and should not be part of the public API which is released under SemVer. We can't release internals under SemVer. Otherwise they're not internals.

What should happen instead is that we have an actual allow list for things people usually import from deep paths but should be imported from public paths. Otherwise we're still transforming everything that's not a component but private.

Reply:
To give some background, I started thinking about this, only because these imports were allowed in the [v4.0.0 `optimal-imports` codemod](https://github.com/mui-org/material-ui/blob/next/packages/material-ui-codemod/src/v4.0.0/optimal-imports.test/actual.js#L60). Currently, this codemod is transforming all imports deeper than 1 level. If we transform these imports, we are basically breaking the build, as these are not exported from their parent's directory. 

Maybe we should not do anything about it, and have the build failed if some of these components were used.

Reply:
A codemod shouldn't break anything. Seems to me `optimal-imports` is already doing everything we want it to do and should just be extended to a v5 version.

truly private imports should just not be touched.

Reply:
Alright, I've adjusted the `optimal-imports` for v5 and exported it under the name `private-imports`. I've removed this array.

Reply:
The only difference between this codemod and the `optimal` imports is that we don't read from the `@material-ui/core/es` anymore, but from `@material-ui/core`